pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -215,42 +234,57 @@ func getPortMap(pod *v1.Pod) map[string]uint32 {
 	return pmap
 }
 
-// deleteIP returns true if the pod and ip are really deleted.
-func (pc *PodCache) deleteIP(ip string, podKey types.NamespacedName) bool {
+// deleteIPs returns true if the pod and ip are really deleted.
+func (pc *PodCache) deleteIPs(ips []string, podKey types.NamespacedName) bool {
 	pc.Lock()
 	defer pc.Unlock()
-	if pc.podsByIP[ip] == podKey {
-		delete(pc.podsByIP, ip)
-		delete(pc.IPByPods, podKey)
-		return true
+	deleteCounter := 0
+	for _, itemIP := range ips {
+		if pc.podsByIP[itemIP] == podKey {
+			delete(pc.podsByIP, itemIP)
+			delete(pc.IPByPods, podKey)
+			deleteCounter++
+		}
 	}
-	return false
+	return deleteCounter == len(ips)
```

Comment:
I don't think this logic is correct. We are returning "were all IPs deleted". But the check is used by callers to mean "was there any change". So it should be "was ANY IP deleted".

Reply:
Yeah, I think this issue can be confirmed if the question [above](https://github.com/istio/istio/pull/44115/files#r1188014465) can be handled.